File: VB/TestMyTreeListForum/MyToolButtonCollection.vb
Imports System
Imports System.Collections.Generic



Namespace DevExpress.MyControl
    Public Class MyToolButtonCollection
        Private buttonList As List(Of MyToolButton)


        Public Sub New()
            buttonList = New List(Of MyToolButton)()
        End Sub



        Public ReadOnly Property Count() As Integer
            Get
                Return buttonList.Count
            End Get
        End Property



        Public Sub Add(ByVal btn As MyToolButton)
            If btn Is Nothing OrElse buttonList.Contains(btn) Then
                Return
            End If
            buttonList.Add(btn)
        End Sub



        Public Sub Remove(ByVal btn As MyToolButton)
            If btn Is Nothing Then
                Return
            End If
            buttonList.Remove(btn)
        End Sub



        Public Sub RemoveAt(ByVal index As Integer)
            If index > buttonList.Count - 1 OrElse index < 0 Then
                Return
            End If

            buttonList.RemoveAt(index)
        End Sub



        Default Public ReadOnly Property Item(ByVal index As Integer) As MyToolButton
            Get
                Return (If(index > buttonList.Count - 1 OrElse index < 0, Nothing, buttonList(index)))
            End Get
        End Property



        Public Sub Clear()
            buttonList.Clear()
        End Sub
    End Class
End Namespace

File: CS/TestMyTreeListForum/MyToolButtonCollection.cs
using System;
using System.Collections.Generic;



namespace DevExpress.MyControl
{
    public class MyToolButtonCollection
    {
        private List<MyToolButton> buttonList;


        public MyToolButtonCollection()
        {
            buttonList = new List<MyToolButton>();
        }



        public int Count
        {
            get
            {
                return buttonList.Count;
            }
        }



        public void Add(MyToolButton btn)
        {
            if (btn == null || buttonList.Contains(btn)) return;
            buttonList.Add(btn);
        }



        public void Remove(MyToolButton btn)
        {
            if (btn == null) return;
            buttonList.Remove(btn);
        }



        public void RemoveAt(int index)
        {
            if (index > buttonList.Count - 1 || index < 0) return;

            buttonList.RemoveAt(index);
        }



        public MyToolButton this[int index]
        {
            get
            {
                return (index > buttonList.Count - 1 || index < 0 ? null : buttonList[index]);
            }
        }



        public void Clear()
        {
            buttonList.Clear();
        }
    }
}
